addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/protocol/utils/VolumeConverter.java

```diff
@@ -15,72 +15,71 @@
  * @author Antoine Besnard - Initial contribution
  */
 public final class VolumeConverter {
-
-    private static final String IP_CONTROL_VOLUME_FORMAT = "000";
-    private static final String IP_CONTROL_VOLUME_DEFAULT_VALUE = "000";
-
-    private static final double MAX_IP_CONTROL_VOLUME = 184;
-    private static final double MIN_DB_VOLUME = 80;
-
-    private static final DecimalFormat FORMATTER = new DecimalFormat(IP_CONTROL_VOLUME_FORMAT);
-
+    
+    private static final String[] IP_CONTROL_VOLUME_FORMAT = { "000", "00", "00", "00" };
+    private static final String[] IP_CONTROL_VOLUME_DEFAULT_VALUE = { "000", "00", "00", "00" };
+    
+    private static final double[] MAX_IP_CONTROL_VOLUME = { 184, 80, 80, 80 };
+    private static final double[] MIN_DB_VOLUME = { 80, 80, 80, 80 };
+    
     /**
      * Return the double value of the volume from the value received in the IpControl response.
      *
      * @param ipControlVolume
      * @return the volume in Db
      */
-    public static double convertFromIpControlVolumeToDb(String ipControlVolume) {
+    public static double convertFromIpControlVolumeToDb(String ipControlVolume, int zone) {
         double ipControlVolumeInt = Double.parseDouble(ipControlVolume);
-        return ((ipControlVolumeInt - 1d) / 2d) - MIN_DB_VOLUME;
+        return ((ipControlVolumeInt - 1d) / 2d) - MIN_DB_VOLUME[zone - 1];
```

Comment:
yes. from 0 to 2 and to 3 when HDZone is introduced with my other Pull Request "[PioneerAvr] Added support for 2014, 2015 and 2016 Lineup of Pioneers AVRs"

`ArrayIndexOutOfBounds` can not completely be ruled out in the current design. But the zone number is extracted from channel-id of openHAB Chanel-group-type which are defined as (zone1Controls - zone4Controls)